pyvo/mivot/features/sky_coord_builder.py

```diff
@@ -0,0 +1,170 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Utility transforming MIVOT annotation into SKYCoord instances
+"""
+
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+from astropy.coordinates import ICRS, Galactic, FK4, FK5
+
+
+class MangoRoles:
+    """
+    Place holder for the roles of the mango:EpochPosition roles
+    """
+    LONGITUDE = "longitude"
+    LATITUDE = "latitude"
+    PM_LONGITUDE = "pmLongitude"
+    PM_LATITUDE = "pmLatitude"
+    PARALLAX = "parallax"
+    RADIAL_VELOCITY = "radialVelocity"
+    EPOCH = "epoch"
+    FRAME = "frame"
+    EQUINOX = "equinox"
+    PMCOSDELTAPPLIED = "pmCosDeltApplied"
+
+
+# Mapping of the MANGO parameters on the SkyCoord parameters
+skycoord_param_default = {
+    MangoRoles.LONGITUDE: 'ra', MangoRoles.LATITUDE: 'dec', MangoRoles.PARALLAX: 'distance',
+    MangoRoles.PM_LONGITUDE: 'pm_ra_cosdec', MangoRoles.PM_LATITUDE: 'pm_dec',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime'}
+
+skycoord_param_galactic = {
+    MangoRoles.LONGITUDE: 'l', MangoRoles.LATITUDE: 'b', MangoRoles.PARALLAX: 'distance',
+    MangoRoles.PM_LONGITUDE: 'pm_l_cosb', MangoRoles.PM_LATITUDE: 'pm_b',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime'}
+
+
+class SkyCoordBuilder(object):
+    '''
+    Utility generating SkyCoord instances from MIVOT annotations
+
+    - SkyCoord instances can only be built from model classes containing the minimal
+    set of required parameters (a position).
+    - In this implementation, only the mango:EpochPosition class is supported since
+      it contains the information required to compute the epoch propagation which is a major use-case
+    '''
+
+    def __init__(self, mivot_instance_dict):
+        '''
+        Constructor
+
+        parameters
+        -----------
+        - mivot_instance_dict: viewer.MivotInstance.to_dict()
+                               Internal dictionary of the dynamic Python object generated
+                               from the MIVOT block
+        '''
+        self._mivot_instance_dict = mivot_instance_dict
+        self._map_coord_names = None
+
+    def build_sky_coord(self):
+        """
+        Build a SkyCoord instance from the dynamic Python object attribute.
+
+        returns
+        -------
+        - SkyCoord instance or None
+        """
+        if self._mivot_instance_dict and self._mivot_instance_dict["dmtype"] == "mango:EpochPosition":
+            return self._build_sky_coord_from_mango()
```

Comment:
I'm wondering if we should throw an exception instead if a user wants to get the sky coordinates but the MIVOT metadata is not available with the results. The advantage is that the message in the exception can state the issue clearly whereas `None` is less informative. Opinions?

Reply:
I've no strong opinion about this. 
Having no `mango:EpochPosition` instance in the mapping is not an error, this why I choose to return `None`, but I agree that using an exception allows to tell the users why he/she cannot get a `SkyCoord` .
If we do so, we cannot use the `MivotException` which is risen when something went wrong with the annotations which is not the case here.
I think that if we decide to use an exception, we should add a new one (e.g. `NoMatchingClass`) or use Python built-in one.

Reply:
After 2 days of thinking, followed by a short discussion with @ManonMarchand, who also suggested a warning, I am leaning towards implementing an exception. 
The reason for this decision is that I bet that similar cases will occur again as long as the binding of PyVO to Mivot gets tighter. So it is wise to foresee a pattern for cases where mapped data does not cover the PYVO API expectations.

The last difficulty is find out a relevant name since the concepts of class/instance also refer to Python stuff. 
I propose : `NoMatchingDMType`.
If none has objection, I'll implement this quickly

Reply:
Done